vehicle_simulation_environment/car.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+import time
+from threading import Timer
+
+
+class Parameters(object):
+    def __init__(
+        self,
+        x_start,
+        y_start,
+        theta_start,
+        vel_start,
+        accel_start,
+        angvel_start,
+        ang_accel_start,
+    ):
+        self.x = x_start
+        self.y = y_start
+        self.theta = theta_start
+        self.vel = vel_start
+        self.accel = accel_start
+        self.ang_vel = angvel_start
+        self.ang_accel = ang_accel_start
+
+
+class Car(object):
+    """Model of a car as a point"""
+
+    def __init__(self, init_param):
+        self.REFRESH_RATE = 2000
+        self.START = 1
+        self.reset(init_param)
+        self.timer = Timer(0, self.update_location)
+        self.timer.start()
+
+    def reset(self, param):
+        self.x = param.x
+        self.y = param.y
+        self.theta = param.theta % (2 * np.pi)
+        self.vel = param.vel
+        self.ang_vel = param.ang_vel
+        self.prev_time = time.monotonic()
+        self.cur_time = time.monotonic()
+        self.accel = 0
+        self.ang_accel = 0
+        self.update_param(param.accel, param.ang_accel)
+
+    def update_param(self, delta_accel, delta_ang_accel):
+        self.accel += delta_accel
+        self.ang_accel += delta_ang_accel
+
+    def stop(self):
+        self.START = 0
+        self.timer.cancel()
+        print(
+            f"""
+            x = {self.x}\n
+            y = {self.y}\n
+            theta = {self.theta}\n
+            velocity = {self.vel}\n
+            ang vel = {self.ang_vel}\n 
+            accel = {self.accel}\n 
+            ang accel = {self.ang_accel}"""
+        )
+
+    def update_location(self):
```

Comment:
Let's have a discussion about this in-person. I'm personally not a fan of this design principle as you're adding some uncertainty to the execution of the program as runs will not be exactly deterministic, something we'd like from a simulation. Instead I believe this class should have some kind of "time step" driven by some external form of program execution.